### PR TITLE
Rearrange hierarchy for Eateries view controller

### PR DESCRIPTION
**Pull request by whoiswillma**
## Overview

Rearrange the view controller hierarchy by adding a navigation controller for both the campus and collegetown eatery VCs. 

This refactoring was done to make way for improved search. 

I spent most of my time dealing with a frustrating UIKit bug where the navigation bar appear compact when the app loaded. If anyone has information about what could be causing this please let me know.

## Changes Made

Originally, the view controller hierarchy looked like this:

- Eatery Navigation VC
  - Eateries Shared VC
    - Pill VC
      - Campus Eatery VC
      - Collegetown Eatery VC

Now, the view controller hierarchy is arranged:

- Eateries Shared VC
  - Pill VC
    - Eatery Navigation VC
      - Campus Eatery VC
    - Eatery Navigation VC
      - Collegetown Eatery VC

## Test Coverage

Manual testing on device

## Related PRs or Issues (delete if not applicable)

#273

### CODE REVIEW

File: Eatery/Controllers/Eateries/Campus/CampusEateriesViewController.swift [hunk 1]
```diff
@@ -90,6 +90,17 @@ class CampusEateriesViewController: EateriesViewController {
         pushPreselectedEateryIfPossible()
     }
 
+    override func pushMapViewController() {
+        super.pushMapViewController()
+
+        guard let eateries = allEateries else {
```
Comment (usubzero) [inline Eatery/Controllers/Eateries/Campus/CampusEateriesViewController.swift]: Nit: one line this guard

File: Eatery/Controllers/Eateries/Collegetown/CollegetownEateriesViewController.swift [hunk 2]
```diff
@@ -48,6 +48,17 @@ class CollegetownEateriesViewController: EateriesViewController {
         }
     }
 
+    override func pushMapViewController() {
+        super.pushMapViewController()
+
+        guard let eateries = allEateries else {
```
Comment (usubzero) [inline Eatery/Controllers/Eateries/Collegetown/CollegetownEateriesViewController.swift]: Nit: one lining guard

File: Eatery/Controllers/Eateries/Collegetown/CollegetownEateriesViewController.swift [hunk 3]
```diff
@@ -48,6 +48,17 @@ class CollegetownEateriesViewController: EateriesViewController {
         }
     }
 
+    override func pushMapViewController() {
```
Comment (usubzero) [inline Eatery/Controllers/Eateries/Collegetown/CollegetownEateriesViewController.swift]: Same code for CollegetownEateries and CampusEateries, is it possible to make eateries part of EateriesViewController and provide an implementation for this function there?

File: Eatery/Controllers/Eateries/EateriesSharedViewController.swift [hunk 4]
```diff
@@ -158,6 +127,13 @@ class EateriesSharedViewController: UIViewController {
         }
     }
 
+    private func updateShowPill() {
+        let showPill = pillViewController.pillView.leftSegmentSelected && campusNavigation.viewControllers.count == 1
```
Comment (usubzero) [inline Eatery/Controllers/Eateries/EateriesSharedViewController.swift]: Nit: clearly operator precedence tells how this is evaluated, but could parenthesis be added around the and clauses for nicer readability?

**Review by usubzero — APPROVED** [review]
Nice job abstracting the pushMapVC fn, all looks good!

File: Eatery/Controllers/Eateries/EateriesSharedViewController.swift [hunk 5]
```diff
@@ -16,25 +16,23 @@ class EateriesSharedViewController: UIViewController {
...
 
-    private(set) lazy var campusEateriesViewController = CampusEateriesViewController()
-    private(set) lazy var collegetownEateriesViewController = CollegetownEateriesViewController()
-
-    private(set) lazy var pillViewController = PillViewController(
-        leftViewController: campusEateriesViewController,
-        rightViewController: collegetownEateriesViewController
+    private lazy var campusEateries = CampusEateriesViewController()
+    private lazy var campusNavigation = EateryNavigationController(rootViewController: campusEateries)
+    private lazy var collegetownEateries = CollegetownEateriesViewController()
```
Comment (kevinchan159) [inline Eatery/Controllers/Eateries/EateriesSharedViewController.swift]: can we add `VC` or something like that to the end of the name just to make it clear

**Review by kevinchan159 — APPROVED** [review]
🔥

**Pull request merged**